dev HEAD: error on `make deploy` when using Python 3.6

```
/opt/local/bin/gmkdir -p Bitcoin-Qt.app/Contents
/opt/local/bin/gmkdir -p Bitcoin-Qt.app/Contents/Resources
/opt/local/bin/gmkdir -p Bitcoin-Qt.app/Contents/Resources
/opt/local/bin/ginstall -c -m 644 src/qt/res/icons/bitcoin.icns Bitcoin-Qt.app/Contents/Resources/bitcoin.icns
/opt/local/bin/gmkdir -p Bitcoin-Qt.app/Contents
/opt/local/bin/ginstall -c -m 644 share/qt/Info.plist Bitcoin-Qt.app/Contents/Info.plist
/Applications/Xcode.app/Contents/Developer/usr/bin/make -C src qt/bitcoin-qt
/opt/local/bin/gmkdir -p Bitcoin-Qt.app/Contents/MacOS
STRIPPROG="/opt/local/bin/strip" /bin/sh /Users/rex/BitcoinUnlimited/build-aux/install-sh -c -s  src/qt/bitcoin-qt Bitcoin-Qt.app/Contents/MacOS/Bitcoin-Qt
/opt/local/bin/gmkdir -p Bitcoin-Qt.app/Contents/Resources/Base.lproj
echo '{	CFBundleDisplayName = "Bitcoin Unlimited Cash Edition"; CFBundleName = "Bitcoin Unlimited Cash Edition"; }' > Bitcoin-Qt.app/Contents/Resources/Base.lproj/InfoPlist.strings
./contrib/macdeploy/macdeployqtplus Bitcoin-Qt.app -add-qt-tr da,de,es,hu,ru,uk,zh_CN,zh_TW -translations-dir= -dmg -fancy ./contrib/macdeploy/fancy.plist -verbose 2 -volname Bitcoin-Unlimited-Cash-Edition
  File "./contrib/macdeploy/macdeployqtplus", line 204
    print "Inspecting with otool: " + binaryPath
                                  ^
SyntaxError: Missing parentheses in call to 'print'. Did you mean print("Inspecting with otool: " + binaryPath)?
make: *** [Bitcoin-Unlimited-Cash-Edition.dmg] Error 1
```

On latest `git pull` after the merge, it now fails with

```
$ make deploy
/Applications/Xcode.app/Contents/Developer/usr/bin/make -C src qt/bitcoin-qt
/opt/local/bin/python3 ./contrib/macdeploy/macdeployqtplus Bitcoin-Qt.app -add-qt-tr da,de,es,hu,ru,uk,zh_CN,zh_TW -translations-dir= -dmg -fancy ./contrib/macdeploy/fancy.plist -verbose 2 -volname Bitcoin-Unlimited-Cash-Edition
+ Removing old dist folder +
+ Copying source bundle +
+ Deploying frameworks +
Traceback (most recent call last):
  File "./contrib/macdeploy/macdeployqtplus", line 625, in <module>
    deploymentInfo = deployFrameworksForAppBundle(applicationBundle, config.strip, verbose)
  File "./contrib/macdeploy/macdeployqtplus", line 367, in deployFrameworksForAppBundle
    frameworks = getFrameworks(applicationBundle.binaryPath, verbose)
  File "./contrib/macdeploy/macdeployqtplus", line 214, in getFrameworks
    otoolLines = o_stdout.split("\n")
TypeError: a bytes-like object is required, not 'str'
make: *** [Bitcoin-Unlimited-Cash-Edition.dmg] Error 1
```

Still getting the same after changing the line:

```
$ make deploy
/Applications/Xcode.app/Contents/Developer/usr/bin/make -C src qt/bitcoin-qt
/opt/local/bin/gmkdir -p Bitcoin-Qt.app/Contents/MacOS
STRIPPROG="/opt/local/bin/strip" /bin/sh /Users/rex/BitcoinUnlimited/build-aux/install-sh -c -s  src/qt/bitcoin-qt Bitcoin-Qt.app/Contents/MacOS/Bitcoin-Qt
/opt/local/bin/python3 ./contrib/macdeploy/macdeployqtplus Bitcoin-Qt.app -add-qt-tr da,de,es,hu,ru,uk,zh_CN,zh_TW -translations-dir= -dmg -fancy ./contrib/macdeploy/fancy.plist -verbose 2 -volname Bitcoin-Unlimited-Cash-Edition
+ Removing old dist folder +
+ Copying source bundle +
+ Deploying frameworks +
Traceback (most recent call last):
  File "./contrib/macdeploy/macdeployqtplus", line 625, in <module>
    deploymentInfo = deployFrameworksForAppBundle(applicationBundle, config.strip, verbose)
  File "./contrib/macdeploy/macdeployqtplus", line 367, in deployFrameworksForAppBundle
    frameworks = getFrameworks(applicationBundle.binaryPath, verbose)
  File "./contrib/macdeploy/macdeployqtplus", line 221, in getFrameworks
    line = line.replace("@loader_path", os.path.dirname(binaryPath))
TypeError: a bytes-like object is required, not 'str'
make: *** [Bitcoin-Unlimited-Cash-Edition.dmg] Error 1
```

I also have multiple Python versions installed.

```
$ python2 --version
Python 2.7.15
$ python3 --version
Python 3.6.7
```

--- a/Makefile.am
+++ b/Makefile.am
@@ -105,8 +105,15 @@ osx_volname:
 	echo $(OSX_VOLNAME) >$@
 
 if BUILD_DARWIN
-$(OSX_DMG): $(OSX_APP_BUILT) $(OSX_PACKAGING)
+$(OSX_DMG): $(OSX_APP_BUILT) $(OSX_PACKAGING) $(OSX_BACKGROUND_IMAGE)
 	$(PYTHON) $(OSX_DEPLOY_SCRIPT) $(OSX_APP) -add-qt-tr $(OSX_QT_TRANSLATIONS) -translations-dir=$(QT_TRANSLATION_DIR) -dmg -fancy $(OSX_FANCY_PLIST) -verbose 2 -volname $(OSX_VOLNAME)
+
+$(OSX_BACKGROUND_IMAGE).png: contrib/macdeploy/$(OSX_BACKGROUND_SVG)
+	sed 's/PACKAGE_NAME/$(PACKAGE_NAME)/' < "$<" | $(RSVG_CONVERT) -f png -d 36 -p 36 -o $@
+$(OSX_BACKGROUND_IMAGE)@2x.png: contrib/macdeploy/$(OSX_BACKGROUND_SVG)
+	sed 's/PACKAGE_NAME/$(PACKAGE_NAME)/' < "$<" | $(RSVG_CONVERT) -f png -d 72 -p 72 -o $@
+$(OSX_BACKGROUND_IMAGE): $(OSX_BACKGROUND_IMAGE).png $(OSX_BACKGROUND_IMAGE)@2x.png
+	tiffutil -cathidpicheck $^ -out $@
 
 deploydir: $(OSX_DMG)
 else

--- a/configure.ac
+++ b/configure.ac
@@ -401,6 +401,7 @@ case $host in
          fi
        fi
 
+       AC_PATH_PROGS([RSVG_CONVERT], [rsvg-convert rsvg],rsvg-convert)
        AC_CHECK_PROG([BREW],brew, brew)
        if test x$BREW = xbrew; then
          dnl These Homebrew packages may be keg-only, meaning that they won't be found
@@ -436,7 +437,6 @@ case $host in
            AC_PATH_TOOL([INSTALLNAMETOOL], [install_name_tool], install_name_tool)
            AC_PATH_TOOL([OTOOL], [otool], otool)
            AC_PATH_PROGS([GENISOIMAGE], [genisoimage mkisofs],genisoimage)
-           AC_PATH_PROGS([RSVG_CONVERT], [rsvg-convert rsvg],rsvg-convert)
            AC_PATH_PROGS([IMAGEMAGICK_CONVERT], [convert],convert)
            AC_PATH_PROGS([TIFFCP], [tiffcp],tiffcp)
 

--- a/contrib/macdeploy/macdeployqtplus
+++ b/contrib/macdeploy/macdeployqtplus
@@ -210,8 +210,8 @@ def getFrameworks(binaryPath, verbose):
             sys.stderr.write(o_stderr)
             sys.stderr.flush()
             raise RuntimeError("otool failed with return code %d" % otool.returncode)
-    
-    otoolLines = o_stdout.split("\n")
+
+    otoolLines = o_stdout.decode().split("\n")
     otoolLines.pop(0) # First line is the inspected binary
     if ".framework" in binaryPath or binaryPath.endswith(".dylib"):
         otoolLines.pop(0) # Frameworks and dylibs list themselves as a dependency.
@@ -676,7 +676,7 @@ if verbose >= 2:
     print("+ Installing qt.conf +")
 
 f = open(os.path.join(applicationBundle.resourcesPath, "qt.conf"), "wb")
-f.write(qt_conf)
+f.write(qt_conf.encode())
 f.close()
 
 # ------------------------------------------------

--- a/contrib/macdeploy/macdeployqtplus
+++ b/contrib/macdeploy/macdeployqtplus
@@ -791,7 +791,7 @@ if config.dmg is not None:
         except subprocess.CalledProcessError as e:
             sys.exit(e.returncode)
         
-        m = re.search("/Volumes/(.+$)", output)
+        m = re.search("/Volumes/(.+$)", output.decode())
         disk_root = m.group(0)
         disk_name = m.group(1)
         
@@ -868,7 +868,7 @@ if config.dmg is not None:
             print(s)
 
         p = subprocess.Popen(['osascript', '-'], stdin=subprocess.PIPE)
-        p.communicate(input=s)
+        p.communicate(input=s.encode('utf-8'))
         if p.returncode:
             print("Error running osascript.")
 

--- a/contrib/macdeploy/macdeployqtplus
+++ b/contrib/macdeploy/macdeployqtplus
@@ -852,7 +852,7 @@ if config.dmg is not None:
             "items_positions" : "\n                   ".join(items_positions)
             }
         if "window_bounds" in fancy:
-            params["window.bounds"] = ",".join([str(p) for p in fancy["window_bounds"]])
+            params["window_bounds"] = ",".join([str(p) for p in fancy["window_bounds"]])
         if "icon_size" in fancy:
             params["icon_size"] = str(fancy["icon_size"])
         if bg_path is not None:

